hir/src/asm/isa.rs

```diff
@@ -616,6 +617,378 @@ pub enum MasmOp {
     /// The behavior is undefined if either `a` or `b` are >= 2^32
     U32UncheckedMax,
 }
+impl MasmOp {
+    pub fn into_node(
+        self,
+        _codemap: &miden_diagnostics::CodeMap,
+        imports: &super::ModuleImportInfo,
+        local_ids: &FxHashMap<FunctionIdent, u16>,
+        proc_ids: &FxHashMap<FunctionIdent, miden_assembly::ProcedureId>,
+    ) -> SmallVec<[miden_assembly::ast::Node; 2]> {
+        use miden_assembly::ast::{Instruction, Node};
+        let node = match self {
+            Self::Padw => Node::Instruction(Instruction::PadW),
```

Comment:
Making `node` to be `Instruction` rather than `Node` would allow us to remove `Node::Instruction(...)` and reduce the case statement to `Self::Padw => Instruction::PadW,`. If `Instruction::*` can be imported, this would even reduce the case to `Self::Padw => PadW,`.

Reply:
I had a similar thought (about handling the `Node`-wrapping at the end of the block), but was already quite some way through the implementation so didn't bother to change it. I can do that before merging though.

As for importing the enum variants directly, I don't like doing that, and prefer to alias the enum type itself if it is too verbose, e.g. `use miden_assembly::ast::Instruction as Inst`, but in this case it's not really that much noise.